### DISCUSSION TITLE
How do I make the actual output the "default" output?

### DISCUSSION THREAD

**Discussion by road2react** [wwmm/easyeffects#2233] [General]
I usually set the default to Speakers:
![image](https://user-images.githubusercontent.com/93143714/229017297-fe3b8b4c-2164-4af3-a5ba-9ea889a5ad41.png)

If I plug in headphones, the default will switch to "Easy Effects Sink"
![image](https://user-images.githubusercontent.com/93143714/229017347-9fb1e77a-e689-43d6-aa46-0627037754f9.png)

I then change it manually, but unplugging also switches it to "Easy Effects Sink" and will have to change it manually again.

When "Easy Effects Sink" is the default, changing volume using volume buttons will not work.

**Comment by wwmm** [comment]
Take a look at #1836 for a more detailed discussion. What these settings are doing is just allowing the user to manually select the device EasyEffects will use for output/input. If you want it to follow your system default device keep `Use Default Input` and `Use Default Output` enabled.

As explained at #1836 unless you have do deal with a very weird corner case that requires our devices to be set as default never do that. Keep you hardware as default.

>I then change it manually, but unplugging also switches it to "Easy Effects Sink" and will have to change it manually again.

 Probably WirePlumber/PipeWire `stream <-> device` restoration code automatically setting our virtual devices as default because that was manually done by the user at some point. Try to delete `~/.local/state/wireplumber/` and reboot so that you reset wireplumber configuration.

**Reply by road2react** [reply]
Thanks for the help!

I deleted `~/.local/state/wireplumber/` and rebooted and the issue doesn't happen anymore. Is there a way to prevent this mistake from occurring in the future?

**Reply by wwmm** [reply]
>Is there a way to prevent this mistake from occurring in the future?

Unfortunately not from our side. There isn't a way for us to tell Pipewire/WirePlumber that our virtual devices should never be used as default.

**Reply by road2react** [reply]
The issue has came back without changing the default device. I'll delete `~/.local/state/wireplumber/` again and see what happens.

Edit: issue (temp) fixed again

Edit 2: I've symlinked `~/.local/state/wireplumber/default-nodes` to `/dev/null`. Hopefully it doesnt happen again.
